docs/openapi-docs/src/main/scala/sttp/tapir/docs/openapi/EndpointToOperationResponse.scala

```diff
@@ -22,90 +21,68 @@ private[openapi] class EndpointToOperationResponse(objectSchemas: Schemas, codec
     }
-
-    if (responses.isEmpty) {
-      // no output at all - using default if defined
-      defaultResponse.map(defaultResponseKey -> Right(_)).toIterable.toListMap
-    } else responses
   }
 
-  private def outputsToResponse(sc: Option[sttp.model.StatusCode], outputs: Vector[EndpointOutput[_]]): Option[Response] = {
-    val headers = outputs.collect {
-      case EndpointIO.Header(name, codec, info) =>
-        name -> Right(
-          Header(
-            info.description,
-            Some(!codec.schema.isOptional),
-            None,
-            None,
-            None,
-            None,
-            None,
-            Some(objectSchemas(codec)),
-            info.example.flatMap(exampleValue(codec, _)),
-            ListMap.empty,
-            ListMap.empty
-          )
-        )
-      case EndpointIO.FixedHeader(h, _, info) =>
-        h.name -> Right(
-          Header(
-            info.description,
-            Some(true),
-            None,
-            None,
-            None,
-            None,
-            None,
-            Option(Right(ASchema(ASchemaType.String))),
-            None,
-            ListMap.empty,
-            ListMap.empty
-          )
-        )
-    }
-
-    val bodies = outputs.collect {
-      case EndpointIO.Body(_, codec, info) => (info.description, codecToMediaType(codec, info.examples))
-      case EndpointIO.StreamBodyWrapper(StreamBodyIO(_, codec, info, _)) =>
-        (info.description, codecToMediaType(codec, info.examples))
-    }
-    val body = bodies.headOption
+  private def outputsToResponse(sc: Option[sttp.model.StatusCode], outputs: List[EndpointOutput[_]]): Option[Response] = {
+    val bodies = collectBodies(outputs)
+    val headers = collectHeaders(outputs)
 
     val statusCodeDescriptions = outputs.flatMap {
       case EndpointOutput.StatusCode(possibleCodes, _, _)                          => possibleCodes.filter(c => sc.contains(c._1)).flatMap(_._2.description)
       case EndpointOutput.FixedStatusCode(_, _, EndpointIO.Info(Some(desc), _, _)) => Vector(desc)
       case _                                                                       => Vector()
     }
 
-    val description = body.flatMap(_._1).getOrElse(statusCodeDescriptions.headOption.getOrElse(""))
+    val description = bodies.headOption.flatMap { case (desc, _) => desc }.getOrElse(statusCodeDescriptions.headOption.getOrElse(""))
 
-    val content = body.map(_._2).getOrElse(ListMap.empty)
+    val content = bodies.flatMap { case (_, content) => content }.toListMap
 
-    if (body.isDefined || headers.nonEmpty) {
+    if (bodies.nonEmpty || headers.nonEmpty) {
       Some(Response(description, headers.toListMap, content))
     } else if (outputs.nonEmpty) {
       Some(Response(description, ListMap.empty, ListMap.empty))
     } else {
       None
     }
   }
+
+  private def collectBodies(outputs: List[EndpointOutput[_]]): List[(Option[String], ListMap[String, MediaType])] = {
+    outputs.collect {
```

Comment:
the bodies can be nested in maps etc., so here we would need to inspect the collection of unwrapped basic outputs

Reply:
there's `output.traverseOutputs` extension method which I think would be better here